webapp/tests/forms/steps/test_eligibility_steps.py

```diff
@@ -82,97 +84,69 @@ def test_if_married_and_joint_taxes_false_then_return_2(self):
-        name = "SECOND_VALID_STEP"
-
-    class InValidPreviousStep(DecisionEligibilityInputFormSteuerlotseStep):
-        name = "INVALID_STEP"
-        invalid_data_model = MagicMock()
-        invalid_data_model.parse_obj = MagicMock(side_effect=ValidationError([], None))
-        data_model = invalid_data_model
-
-    def test_if_no_previous_step_then_do_not_change_prev_url(self):
-        stored_data = {}
-        previous_url = "PREVIOUS_URL"
-        with app.app_context() and app.test_request_context():
-            step = EligibilityInputFormSteuerlotseStep(endpoint="lotse")
-            step.name = "CURRENT"
-            step._pre_handle()
-            step.render_info.prev_url = previous_url
-            step.previous_steps = None
-
-            step.set_correct_previous_link(stored_data)
-
-            self.assertEqual(previous_url, step.render_info.prev_url)
-
-    def test_if_one_prev_step_then_set_prev_url_to_that_step(self):
-        stored_data = {}
-        with app.app_context() and app.test_request_context():
-            step = EligibilityInputFormSteuerlotseStep(endpoint="lotse")
-            previous_url = step.url_for_step(self.InValidPreviousStep.name)
-            step.name = "CURRENT"
-            step._pre_handle()
-            step.render_info.prev_url = previous_url
-            step.previous_steps = [self.InValidPreviousStep]
-
-            step.set_correct_previous_link(stored_data)
-
-            self.assertEqual(previous_url, step.render_info.prev_url)
-
-    def test_if_invalid_and_valid_are_in_prev_step_then_set_prev_url_to_valid_step(self):
-        stored_data = {}
-        with app.app_context() and app.test_request_context():
-            step = EligibilityInputFormSteuerlotseStep(endpoint="lotse")
-            previous_url = step.url_for_step(self.ValidPreviousStep.name)
-            step.name = "CURRENT"
-            step._pre_handle()
-            step.render_info.prev_url = previous_url
-            step.previous_steps = [self.InValidPreviousStep, self.ValidPreviousStep]
-
-            step.set_correct_previous_link(stored_data)
-
-            self.assertEqual(previous_url, step.render_info.prev_url)
-
-    def test_if_invalid_and_valid_and_second_valid_are_in_prev_step_then_set_prev_url_to_valid_step(self):
-        stored_data = {}
-        with app.app_context() and app.test_request_context():
-            step = EligibilityInputFormSteuerlotseStep(endpoint="lotse")
-            previous_url = step.url_for_step(self.ValidPreviousStep.name)
-            step.name = "CURRENT"
-            step._pre_handle()
-            step.render_info.prev_url = previous_url
-            step.previous_steps = [self.InValidPreviousStep, self.ValidPreviousStep, self.SecondValidPreviousStep]
-
-            step.set_correct_previous_link(stored_data)
-
-            self.assertEqual(previous_url, step.render_info.prev_url)
+class TestEligibilityInputFormSteuerlotseStepIsPreviousStep(unittest.TestCase):
```

Comment:
What about two valid models (with one matching the step name or both not matching)? This would be more for documentation reasons and to know what should happen then. I know that in actual use this probably will never happen

Reply:
Exactly. that's why I didn't add them. But sure, can do.

Reply:
Did it^^